snuba/subscriptions/worker.py

```diff
@@ -180,6 +180,19 @@ def callback_func(
                         },
                     )
 
+                    # Only log 1 in 100 mistmatches to sentry
+                    if match is False and random.random() < 0.01:
+                        logger.warning(
+                            "Non matching result with consistent and non-consistent subscription query",
+                            extra={
+                                "dataset": self.__dataset_name,
+                                "consistent": str(is_consistent_query),
+                                "primary": primary_result,
+                                "other": result.result,
+                                "query": request.query,
```

Comment:
This is the query object that does not have a pretty representation. I think you will just see the object id. 
If you want a formatted query you would need to pass this through a formatter as we were doing for the errors migration.